#1. Create a GFF file for use with DEXSeq
cd /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/refs
wget ftp://ftp.ensembl.org/pub/release-105/gtf/homo_sapiens/Homo_sapiens.GRCh38.105.gtf.gz
gunzip Homo_sapiens.GRCh38.105.gtf.gz

isub -m 10 -i 'docker(malachig/htseq-0.12.4)'
python /usr/local/DEXSeq/dexseq_prepare_annotation.py --aggregate no /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/refs/Homo_sapiens.GRCh38.105.gtf /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff
exit

#2. Create annotations from the GFF to help annotate the practically useless exon count file that will be produced in the next step
cd /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/
grep -v aggregate_gene Homo_sapiens.GRCh38.105.gff | perl -ne 'chomp; $gid="ZZZ"; $eid="ZZZ"; if ($_ =~ /exonic\_part\_number\s+\"(\w+)\".*gene\_id\s+\"(\w+)\"/){$eid=$1; $gid=$2;} print "$gid:$eid\t$_\n"' | sort > Homo_sapiens.GRCh38.105.annotations.tsv

#NOTE: Initially I was NOT understanding that the DEXseq package in R will reassociate counts with annotations using the GFF file.  So the step above is probably not needed.

#3. Prepare exon counts using DEXseq-count and HTSeq-count
#you need to inform the script about unstranded data by specifying the option -s no. If your library preparation protocol reverses the strand (i.e., reads appear on the strand opposite to their gene of origin), use -s reverse. In case of paired-end data, the default (-s yes) means that the read from the first sequence pass is on the same strand as the gene and the read from the second pass on the opposite strand (“forward-reverse” or “fr” order in the parlance of the Bowtie/TopHat manual) and the options -s reverse specifies the opposite case.

#details from the collaborator on library prep: ...

#contruct exon counting commands, one for each sample BAM file

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_cd_1.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div6_cd_1.AACTCCGATC-ATCCGTTGGC/div6_cd_1.AACTCCGATC-ATCCGTTGGC.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_cd_1.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_cd_2.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div6_cd_2.GTCCGAGGAA-CTTCAGGTAA/div6_cd_2.GTCCGAGGAA-CTTCAGGTAA.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_cd_2.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_cd_3.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div6_cd_3.CGACGTACAC-TTACGTTCTC/div6_cd_3.CGACGTACAC-TTACGTTCTC.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_cd_3.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_wt_1.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div6_wt_1.AAGTATCACG-ATCCTCTTGG/div6_wt_1.AAGTATCACG-ATCCTCTTGG.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_wt_1.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_wt_2.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div6_wt_2.AGGTGAGAGC-CTGCACACTG/div6_wt_2.AGGTGAGAGC-CTGCACACTG.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_wt_2.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_wt_3.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div6_wt_3.AATTGGCACA-CTTGGAGCGA/div6_wt_3.AATTGGCACA-CTTGGAGCGA.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div6_wt_3.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_cd_1.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div7_cd_1.AGACCGAATG-CTGGTCGTTC/div7_cd_1.AGACCGAATG-CTGGTCGTTC.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_cd_1.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_cd_2.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div7_cd_2.AGGTCAGCGA-TACACGAAGA/div7_cd_2.AGGTCAGCGA-TACACGAAGA.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_cd_2.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_cd_3.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div7_cd_3.CAGTTCATGG-CTTGGACCTG/div7_cd_3.CAGTTCATGG-CTTGGACCTG.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_cd_3.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_wt_1.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div7_wt_1.CCGACACATC-GACATGTAGA/div7_wt_1.CCGACACATC-GACATGTAGA.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_wt_1.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_wt_2.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div7_wt_2.CGATCAGAAG-AGTAATCGAC/div7_wt_2.CGATCAGAAG-AGTAATCGAC.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_wt_2.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_wt_3.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div7_wt_3.GTATGGCTAA-GCCAGCTATC/div7_wt_3.GTATGGCTAA-GCCAGCTATC.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div7_wt_3.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_cd_1.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div8_cd_1.CCTAAGCTGA-ACCAGATTCA/div8_cd_1.CCTAAGCTGA-ACCAGATTCA.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_cd_1.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_cd_2.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div8_cd_2.TCAGTCGGCT-GCAGATGTAA/div8_cd_2.TCAGTCGGCT-GCAGATGTAA.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_cd_2.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_cd_3.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div8_cd_3.AATCCGAGTA-TTAGAATCGG/div8_cd_3.AATCCGAGTA-TTAGAATCGG.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_cd_3.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_wt_1.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div8_wt_1.CCATCTGTCG-CTCTCATGCG/div8_wt_1.CCATCTGTCG-CTCTCATGCG.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_wt_1.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_wt_2.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div8_wt_2.AGATCGCTCG-GACGCATTGA/div8_wt_2.AGATCGCTCG-GACGCATTGA.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_wt_2.exon_counts.tsv'

bsub -G compute-oncology -q oncology -g /mgriffit/small -M 20000000 -R 'select[mem>20000] span[hosts=1] rusage[mem=20000]' -oo /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_wt_3.exon_counts.log -a 'docker(malachig/htseq-0.12.4)' 'python /usr/local/DEXSeq/dexseq_count.py --stranded no --paired yes --format bam --order pos /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/Homo_sapiens.GRCh38.105.gff /storage1/fs1/alberthkim/Active/data/hao/htcf.wustl.edu/files/pd6ZzJdl/Kim_s5703_MGI2530/div8_wt_3.CCGTGAACTG-TATGCAACTC/div8_wt_3.CCGTGAACTG-TATGCAACTC.genome_accepted_hits.bam /storage1/fs1/alberthkim/Active/data/hao/dexseq_analysis/exon_counts/not_aggregated/div8_wt_3.exon_counts.tsv'

#Create clean versions of the DEXseq count files without the statistics lines at the bottom
cd /storage1/fs1/mgriffit/Active/kommagani_SplicingAnalysis/exon_counts/not_aggregated/
grep -v "^_" div6_cd_1.exon_counts.tsv > div6_cd_1.exon_counts.clean.tsv
grep -v "^_" div6_cd_2.exon_counts.tsv > div6_cd_2.exon_counts.clean.tsv
grep -v "^_" div6_cd_3.exon_counts.tsv > div6_cd_3.exon_counts.clean.tsv
grep -v "^_" div6_wt_1.exon_counts.tsv > div6_wt_1.exon_counts.clean.tsv
grep -v "^_" div6_wt_2.exon_counts.tsv > div6_wt_2.exon_counts.clean.tsv
grep -v "^_" div6_wt_3.exon_counts.tsv > div6_wt_3.exon_counts.clean.tsv
grep -v "^_" div7_cd_1.exon_counts.tsv > div7_cd_1.exon_counts.clean.tsv
grep -v "^_" div7_cd_2.exon_counts.tsv > div7_cd_2.exon_counts.clean.tsv
grep -v "^_" div7_cd_3.exon_counts.tsv > div7_cd_3.exon_counts.clean.tsv
grep -v "^_" div7_wt_1.exon_counts.tsv > div7_wt_1.exon_counts.clean.tsv
grep -v "^_" div7_wt_2.exon_counts.tsv > div7_wt_2.exon_counts.clean.tsv
grep -v "^_" div7_wt_3.exon_counts.tsv > div7_wt_3.exon_counts.clean.tsv
grep -v "^_" div8_cd_1.exon_counts.tsv > div8_cd_1.exon_counts.clean.tsv
grep -v "^_" div8_cd_2.exon_counts.tsv > div8_cd_2.exon_counts.clean.tsv
grep -v "^_" div8_cd_3.exon_counts.tsv > div8_cd_3.exon_counts.clean.tsv
grep -v "^_" div8_wt_1.exon_counts.tsv > div8_wt_1.exon_counts.clean.tsv
grep -v "^_" div8_wt_2.exon_counts.tsv > div8_wt_2.exon_counts.clean.tsv
grep -v "^_" div8_wt_3.exon_counts.tsv > div8_wt_3.exon_counts.clean.tsv
wc -l *.clean.tsv *annotations.tsv


